src/components/LoadDataset.js

```diff
@@ -31,56 +79,126 @@ const LoadDataset = () => {
       (event, filePaths) => {
         if (filePaths && filePaths.length) {
           // currently we select only one file
-          handleFileLocation(filePaths[0]);
+          this.handleLocationInput(filePaths[0]);
         }
       },
     );
   };
 
-  const handleCopy = () => {
-    window.ipcRenderer.send(LOAD_DATASET_CHANNEL, { fileLocation });
+  handleFileSubmit = () => {
+    const { fileCustomName, fileLocation, fileDescription } = this.state;
+    const { dispatchAddDataset } = this.props;
+    dispatchAddDataset({ fileCustomName, fileLocation, fileDescription });
   };
 
-  return (
-    <Main fullScreen>
-      <FormControl>
-        <Typography variant="h4" color="inherit" style={{ margin: '2rem' }}>
-          {t('Load a File')}
-        </Typography>
-        <Button
-          // id={LOAD_BROWSE_BUTTON_ID}
-          variant="contained"
-          onClick={handleBrowse}
-          color="primary"
-          // className={classes.button}
-        >
-          {t('Browse')}
-        </Button>
-        <Input
-          // id={LOAD_INPUT_ID}
-          required
-          onChange={handleFileLocation}
-          // className={classes.input}
-          // inputProps={{
-          //   'aria-label': 'Description',
-          // }}
-          autoFocus
-          value={fileLocation}
-          type="text"
-        />
-        <Button
-          //   id={LOAD_SUBMIT_BUTTON_ID}
-          variant="contained"
-          onClick={handleCopy}
-          color="primary"
-          //   className={classes.button}
-          disabled={!fileLocation.endsWith('.json')}
-        >
-          {t('Submit')}
-        </Button>
-      </FormControl>
-    </Main>
-  );
+  render() {
+    const { classes, t, open, handleClose } = this.props;
+    const { fileLocation, fileCustomName, fileDescription } = this.state;
+
+    return (
+      <Dialog
+        open={open}
+        onClose={() => {
+          this.setState({
+            fileLocation: '',
+            fileCustomName: '',
+            fileDescription: '',
+          });
+          handleClose();
+        }}
+        maxWidth="sm"
+        fullWidth
+      >
+        <DialogTitle id="create-new-item-form">
+          {t('Add new dataset')}
+        </DialogTitle>
+        <DialogContent className={classes.dialogContent}>
+          <div className={classes.addDataset}>
+            <TextField
+              autoFocus
+              margin="dense"
+              id="browse"
+              label="Select dataset"
+              className={classes.shortTextfield}
+              onChange={this.handleLocationInput}
+              value={fileLocation}
+            />
+            <IconButton onClick={this.handleBrowse}>
+              <SearchIcon />
+            </IconButton>
+          </div>
+
+          <TextField
+            margin="dense"
+            id="name"
+            label="Name"
```

Comment:
Translate. I would maybe go for `Dataset Name` or something similar as `Name` is rather vague.